### ISSUE TITLE
signalr-client "normalizes" network-path references that start with "//".

### ISSUE THREAD

**Issue by 686d7066** [aspnet/SignalR#1436]
We have a website and multiple endpoints that provide SignalR hubs.
e.g. the website is served from localhost:5000 and needs to be able to connect to a hub on localhost:5001.

Since the endpoint is not the same as the URL the website was served from we can't simply provide the HubName to the HubConnection. So we provide the URL as a network-path reference, because the website currently must support HTTP and HTTPS. 
So we pass '//localhost:5001/signalr/NotificationHub' to the HubConnection.
We are receiving the following message in the browsers console:

> Information: Normalizing '//localhost:5001/signalr/NotificationHub' to 'http://localhost:5001//localhost:5001/signalr/NotificationHub'

This happens with the signalr-client package with version 1.0.0-alpha2-final.

The client should not normalize URLs beginning with // by adding protocol AND host (again) and instead let the browser handle the network-path reference (to keep in line with https://tools.ietf.org/html/rfc3986#section-4.2) or alternatively only put the current protocol in front.

**Comment by anurse** [comment]
Seems legit. Also fairly straightforward. If you're at all interested in contributing, we'd certainly take a PR that fixes this! We'll put it in our queue as well though.

**Comment by pardahlman** [comment]
@anurse I'd be happy to try to help out here. The only thing I'm uncertain about is how I should go about testing this change... `resolveUrl`, `baseUrl` and `url` are all private. Any ideas?

**Comment by anurse** [comment]
I think we'd do it through mocking our HttpClient interface, like we do with the other HubConnection tests in the TS client. They're a little rough, but you could probably copy one of the others. We just need to verify that the URL that goes to the HttpClient and then we can ignore remaining errors.

**Comment by Wedvich** [comment]
As a dumb temporary workaround, we monkeypatched SignalR in our client app like this, which makes it work as expected for different HTTP/HTTPS endpoints:
```js
import * as signalR from '@aspnet/signalr';

const originalResolveUrl = signalR.HttpConnection.prototype.resolveUrl;
signalR.HttpConnection.prototype.resolveUrl = url =>
  originalResolveUrl(url.indexOf('//') === 0 ? window.document.location.protocol + url : url);
```
Looking forward to a _real_ fix though :smile:

**Comment by anurse** [comment]
Side question (we're still planning to fix this, I'm just wondering): Why are you using a protocol-relative URL? The generally accepted practice these days is to always prefer HTTPS for subresources, even when accessing the main resource over HTTP. In fact, it's becoming generally accepted practice to always redirect HTTP connections to HTTPS. I'm curious if there's a limitation that's preventing you from doing this.

**Comment by xdvarpunen** [comment]
In webpack there is proxy feature in dev server:
```
proxy: {
  '/randomsignalrpath': {
    target: 'ws://localhost:12345',
    ws: true,
  },
}
```
`socket.io` is already using this feature. Basically when I set path, webpack dev server will change ws current path to different path. I wish signalr uses this approach too 👍

**Comment by mikaelm12** [comment]
Fixed in e51852d0fc91c10b8eac641d137d10c9348589ec
